src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileAuthzStep.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.filedata.flight.FileMapKeys;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.service.iam.AuthenticatedUserRequest;
+import bio.terra.service.iam.IamRole;
+import bio.terra.service.iam.IamService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+
+public class IngestFileAuthzStep implements Step {
+    private static final Set<IamRole> readerRoles = EnumSet.of(
```

Comment:
Is there any chance this could be causing serialization issues for the step?

Reply:
I don't think so.